main.go

```diff
@@ -249,6 +253,103 @@ func genMobileStubs(param, godoc map[string]string,
 	}
 }
 
+func genJSStubs(param map[string]string, req *plugin.CodeGeneratorRequest,
+	reg *descriptor.Registry) {
+
+	// We need package_name and target_package in order to continue.
+	pkg := param["package_name"]
+	if pkg == "" {
+		log.Fatal("package name not set")
+	}
+
+	buildTags := param["build_tags"]
+	manualImport := param["manual_import"]
+
+	apiPrefix := false
+	if param["api_prefix"] == "1" {
+		apiPrefix = true
+	}
+
+	for _, filename := range req.FileToGenerate {
+		target, err := reg.LookupFile(filename)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// For each service, we'll create a file with the generated api.
+		for _, s := range target.Services {
+			name := s.GetName()
+			n := strings.ToLower(name)
+
+			f, err := os.Create("./" + n + ".pb.json.go")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			wr := bufio.NewWriter(f)
+
+			// Create the file header.
+			params := jsHeaderParams{
+				ToolName:     versionString,
+				FileName:     filename,
+				ServiceName:  name,
+				Package:      pkg,
+				ManualImport: manualImport,
+				BuildTags:    buildTags,
+			}
+
+			// Go through each method defined by the service, and
+			// call the appropriate template, depending on the RPC
+			// type.
+			for _, m := range s.Methods {
+				name := m.GetName()
+
+				inputType := m.GetInputType()[1:]
```

Comment:
What's sliced off here as part or the first char?

Reply:
a dot `.`

Reply:
Exactly, the types are parsed here as `.lnrpc.GetInfoRequest` for example, we just strip the first dot. Added a comment.